qemu/tests/qemu_guest_agent.py

```diff
@@ -1458,6 +1458,63 @@ def gagent_check_frozen_io(self, test, params, env):
 
         self.gagent_verify(self.params, self.vm)
 
+    @error_context.context_aware
+    def gagent_check_vss_status(self, test, params, env):
+        """
+        Only for windows guest,check QEMU Guest Agent VSS Provider service start type
+        and if it works.
+
+        Steps:
+        1) Check VSS Provider service start type.
+        2) Check VSS Provider service should be in stopped status.
+        3) Issue fsfreeze qga command.
+        4) Check VSS Provider service should be in running status.
+        5) Issue fsthaw qga command.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+        def check_vss_status(status_expect):
+            error_context.context("Check VSS Provider status.", logging.info)
+            cmd_vss_status = "sc query \"QEMU Guest Agent VSS Provider\" | findstr /i STATE"
+            status, output = session.cmd_status_output(cmd_vss_status)
+            if status:
+                test.error("Command to check VSS service status failed,"
+                           "detailed info is:\n%s" % output)
+            vss_status = output.split()[-1]
+            if vss_status != status_expect:
+                test.fail("The output is %s which is not expected."
+                          % vss_status)
+
+        session = self._get_session(self.params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Check VSS Provider service start type.",
+                              logging.info)
+        cmd_vss_start_type = "sc qc \"QEMU Guest Agent VSS Provider\""
+        cmd_vss_start_type += " | findstr /i START_TYPE"
+        status, output = session.cmd_status_output(cmd_vss_start_type)
+        if status:
+            test.error("Command to check VSS service start type failed,"
+                       "detailed info is:\n%s" % output)
+        vss_start_type = output.split()[-1]
+        if vss_start_type != "DEMAND_START":
+            test.fail("The output is %s which is not correct."
+                      % vss_start_type)
```

Comment:
Line 1495 - Line 1504 could be abstracted into function `check_vss_status`, assuming to define it `check_vss_info(command, key, expect_value)`, calling it like `check_vss_info('query', 'STATE', 'STOPPED')` or `check_vss_info('qc', 'START_TYPE', 'DEMAND_START')`.